app/javascript/packages/document-capture/higher-order/with-background-encrypted-upload.jsx

```diff
@@ -2,6 +2,21 @@ import { useContext } from 'react';
 import UploadContext from '../context/upload';
 import AnalyticsContext from '../context/analytics';
 
+/**
+ * @typedef {import('../components/form-steps').FormStepComponentProps<V>} FormStepComponentProps
+ * @template V
+ */
+
+/**
+ * An error representing a failure to complete encrypted upload of image.
+ */
+export class BackgroundEncryptedUploadError extends Error {
+  baseField = '';
+
+  /** @type {string[]} */
+  fields = [];
```

Comment:
would it be worth our time to use the type system to restrict which fields these can be? `front` `back` `front_image_url` `back_image_url` etc etc etc via like a Record type?

Reply:
> would it be worth our time to use the type system to restrict which fields these can be? `front` `back` `front_image_url` `back_image_url` etc etc etc via like a Record type?

We could. As implemented currently, the `withBackgroundEncryptedUpload` doesn't really know anything about the specific fields it's being applied to, so in that sense a generic `string[]` makes sense. It'd be nice if we could be a bit more strict here at least as far as knowing that it'd be one "base" key, then two additional keys `${key}_image_iv` and `${key}_image_url`. Not sure if this is something we can really add type-checking for, though.